notebooks/systemds_dev.ipynb

```diff
@@ -0,0 +1,882 @@
+        "  run(f\"wget -q -nc -O apache-maven.zip https://downloads.apache.org/maven/maven-3/3.6.3/binaries/{maven_version}-bin.zip\")\n",
+        "  run('unzip -q -d /opt apache-maven.zip')\n",
+        "  run('rm -f apache-maven.zip')\n",
+        "\n",
+        "# Let's choose the absolute path instead of $PATH environment variable.\n",
+        "def maven(args):\n",
+        "  run(f\"{maven_path}/bin/mvn {args}\")\n",
+        "\n",
+        "maven('-v')\n"
+      ],
+      "execution_count": 4,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            ">> wget -q -nc -O apache-maven.zip https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.zip\n",
+            "\n",
+            ">> unzip -q -d /opt apache-maven.zip\n",
+            "\n",
+            ">> rm -f apache-maven.zip\n",
+            "\n",
+            ">> /opt/apache-maven-3.6.3/bin/mvn -v\n",
+            "\u001b[1mApache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)\u001b[m\n",
+            "Maven home: /opt/apache-maven-3.6.3\n",
+            "Java version: 1.8.0_252, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre\n",
+            "Default locale: en_US, platform encoding: UTF-8\n",
+            "OS name: \"linux\", version: \"4.19.104+\", arch: \"amd64\", family: \"unix\"\n",
+            "\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Xphbe3R43XLw",
+        "colab_type": "text"
+      },
+      "source": [
+        "#### Download Apache Spark"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_WgEa00pTs3w",
+        "colab_type": "text"
+      },
+      "source": [
+        "NOTE: If spark is not downloaded. Let us make sure the version we are trying to download is officially supported at\n",
+        "https://spark.apache.org/downloads.html"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3zdtkFkLnskx",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 119
+        },
+        "outputId": "a5b8b7ef-a401-4549-e521-320f9f22ed41"
+      },
+      "source": [
+        "# Spark and Hadoop version\n",
+        "spark_version = 'spark-2.4.6'\n",
+        "hadoop_version = 'hadoop2.7'\n",
+        "spark_path = f\"/opt/{spark_version}-bin-{hadoop_version}\"\n",
+        "if not os.path.exists(spark_path):\n",
+        "  run(f\"wget -q -nc -O apache-spark.tgz https://downloads.apache.org/spark/{spark_version}/{spark_version}-bin-{hadoop_version}.tgz\")\n",
+        "  run('tar zxf apache-spark.tgz -C /opt')\n",
+        "  run('rm -f apache-spark.tgz')\n",
+        "\n",
+        "os.environ[\"SPARK_HOME\"] = spark_path\n",
+        "os.environ[\"PATH\"] += \":$SPARK_HOME/bin\"\n"
```

Comment:
I think a version using spark, and one without would be better. Since adding the spark install, unnecessarily complicates this notebook. Also using spark inside this notebook should lead to no improved performance only overhead, since it is executed in a single node.

Reply:
Yes, for now I am adding this as an optional.